signer/tests/integration/transaction_signer.rs

```diff
@@ -359,3 +363,91 @@ async fn handle_pending_deposit_request_address_script_pub_key() {
 
     testing::storage::drop_db(db).await;
 }
+
+/// Test that [`TxSignerEventLoop::assert_valid_stacks_tx_sign_request`]
+/// errors when the signer is not in the signer set.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn signing_set_validation_check_for_stacks_transactions() {
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_emily_client()
+        .with_mocked_stacks_client()
+        .build();
+    let (rpc, faucet) = sbtc::testing::regtest::initialize_blockchain();
+
+    // This confirms a deposit transaction, and has a nice helper function
+    // for storing a real deposit.
+    let setup = TestSweepSetup::new_setup(rpc, faucet, 10000, &mut rng);
+
+    // Let's get the blockchain data into the database.
+    let chain_tip: BitcoinBlockHash = setup.sweep_block_hash.into();
+    backfill_bitcoin_blocks(&db, rpc, &chain_tip).await;
+
+    // This is all normal happy path things that need to happen in order to
+    // pass validation.
+    setup.store_deposit_tx(&db).await;
+    setup.store_sweep_tx(&db).await;
+    setup.store_dkg_shares(&db).await;
+    setup.store_deposit_request(&db).await;
```

Comment:
Naive question; is this going to be necessary for a class of tests, or is this unique to this one? The comment makes me think it'll be necessary for a class of tests, at which point I think this is a good contender to be in something like a test setup function.

Reply:
Done in https://github.com/stacks-network/sbtc/pull/694/commits/4870b46563051b1fcf6805c1780cfd37f0bfa7a2.